### PR TITLE
Fix typo in in-app marketplace sections conditional

### PR DESCRIPTION
**Pull request by corsonr**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

On the **WooCommerce > Marketplace > Shipping** page, the conditional statement that checks the current section has a typo and there's no specific class for the gateways section.

Also, the list of countries where WooCommerce Payments is available has been updated.

This PR addresses those issues.

Closes 11246-gh-Automattic/woocommerce.com

### How to test the changes in this Pull Request:

- Check out this branch on your local environment
- Update the store country / state to any US state from the [settings](https://woocommerce.test/wp-admin/admin.php?page=wc-settings)
- Visit the in-app marketplace [page](https://woocommerce.test/wp-admin/admin.php?page=wc-addons&section=shipping-methods)
- Go to the "Shipping" section from the dropdown select
- Confirm you see the featured "WooCommerce Shipping" large card (see screenshot below)
- Please test around this issue

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> fix in-app marketplace conditional statements

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: includes/admin/meta-boxes/views/html-product-data-general.php [hunk 1]
```diff
@@ -31,8 +31,6 @@
 				'description' => __( 'This text will be shown on the button linking to the external product.', 'woocommerce' ),
 			)
 		);
-		
-		do_action( 'woocommerce_product_options_external' );
```
Comment (roykho) [inline includes/admin/meta-boxes/views/html-product-data-general.php]: Hi @corsonr - what is the reason for removing this hook?

Comment (corsonr) [reply includes/admin/meta-boxes/views/html-product-data-general.php]: Hey @roykho sorry, it seems that there was a conflict. That looks good now.

File: includes/admin/views/html-admin-page-addons.php [hunk 2]
```diff
@@ -77,20 +77,20 @@ class="<?php echo $current_section === $section->slug ? 'current' : ''; ?>"
...
 			<?php if ( '_featured' !== $current_section && $addons ) : ?>
-				<?php if ( 'shipping_methods' === $current_section ) : ?>
+				<?php if ( 'shipping-methods' === $current_section ) : ?>
 					<div class="addons-shipping-methods">
 						<?php WC_Admin_Addons::output_wcs_banner_block(); ?>
 					</div>
 				<?php endif; ?>
 				<?php if ( 'payment-gateways' === $current_section ) : ?>
-					<div class="addons-shipping-methods">
+					<div class="addons-gateways">
```
Comment (andfinally) [inline includes/admin/views/html-admin-page-addons.php]: It's a bit confusing, but the original `addons-shipping-methods` was intentional: we have this style in `admin.scss` which applies to this bit:

```scss
	.addons-shipping-methods .addons-wcs-banner-block {
		margin-left: 0;
		margin-right: 0;
		margin-top: 1em;
	}
```

We could leave it as it is, or we could change this to `addons-payment` and tweak the CSS rule to take the new class into account. Personally I find the reference to the category in`addons-gateways` isn't obvious enough.

**Review by andfinally — COMMENTED** [review]
Thanks for tackling this @corsonr! Apart from the original issue, another requirement has just popped up which I'd love to include in this PR, if possible.

WooCommerce Payments is now available in more countries than the US. We'd like to update the method which adds this banner, `WC_Admin_Addons::output_wcpay_banner_block`, to include the extra countries.

The country list from the sidebar on the [product page](https://woocommerce.com/products/woocommerce-payments/) is

🇦🇺 Australia
🇦🇹 Austria
🇧🇪 Belgium
🇨🇦 Canada
🇫🇷 France
🇩🇪 Germany
🇭🇰 Hong Kong
🇮🇪 Ireland
🇮🇹 Italy
🇳🇱 Netherlands
🇳🇿 New Zealand
🇵🇱 Poland
🇵🇹 Portugal
🇸🇬 Singapore
🇪🇸 Spain
🇨🇭 Switzerland
🇬🇧 United Kingdom
🇺🇸 United States

I'll add a note about this to the [issue](11246-gh-Automattic/woocommerce.com). Please give me a shout if you need more detail. Thanks!

File: includes/admin/views/html-admin-page-addons.php (hunk 2 above)
Comment (corsonr) [reply includes/admin/views/html-admin-page-addons.php]: That makes sense, I adjusted the class name to `addons-payment` and updated `admin.scss`.

**Comment by corsonr** [comment]
I'm closing this PR in favor of https://github.com/woocommerce/woocommerce/pull/30785 (the reason being my forked repo was messed up, head was master instead of trunk)

**Pull request closed without merging**